File: documentation-examples-java/src/java/dev/typr/foundations/docs/core/TransactorSetup.java
package dev.typr.foundations.docs.core;

import dev.typr.foundations.Transactor;
import dev.typr.foundations.connect.*;

@SuppressWarnings("unused")
public class TransactorSetup {
  // start
  // PostgreSQL
  Transactor pgTx =
      Transactor.create(
          PgConfig.builder("localhost", 5432, "mydb", "user", "pass")
              .sslmode(PgSslMode.REQUIRE)
              .build());

  // DuckDB (in-memory)
  Transactor duckTx = Transactor.create(DuckDbConfig.inMemory().build());

  // DuckDB (file-based)
  Transactor duckFileTx =
      Transactor.create(
          DuckDbConfig.builder("/tmp/analytics.db").threads(4).memoryLimit("2GB").build());

  // MariaDB / MySQL
  Transactor mariaTx =
      Transactor.create(MariaConfig.builder("localhost", 3306, "mydb", "user", "pass").build());

  // Oracle
  Transactor oracleTx =
      Transactor.create(
          OracleConfig.builder("localhost", 1521, "xe", "user", "pass")
              .serviceName("XEPDB1")
              .build());

  // SQL Server — .encrypt(FALSE) is correct for local dev against the default container image
  // (self-signed cert, no trust chain). Production should use .encrypt(TRUE) with a trusted
  // certificate; the handshake failure against localhost is otherwise inscrutable.
  Transactor mssqlTx =
      Transactor.create(
          SqlServerConfig.builder("localhost", 1433, "mydb", "user", "pass")
              .encrypt(SqlServerEncrypt.FALSE)
              .build());

  // DB2
  Transactor db2Tx =
      Transactor.create(Db2Config.builder("localhost", 50000, "mydb", "user", "pass").build());
  // stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/core/TransactorSetup.scala
package dev.typr.foundationssc.docs.core
import dev.typr.foundationssc.Transactor
import dev.typr.foundationssc.connect.*

@SuppressWarnings(Array("unused"))
object TransactorSetup:
  // start
  // PostgreSQL
  val pgTx =
    Transactor.create(
      PgConfig
        .builder("localhost", 5432, "mydb", "user", "pass")
        .sslmode(PgSslMode.REQUIRE)
        .build()
    )

  // DuckDB (in-memory)
  val duckTx =
    Transactor.create(DuckDbConfig.inMemory().build())

  // DuckDB (file-based)
  val duckFileTx =
    Transactor.create(
      DuckDbConfig
        .builder("/tmp/analytics.db")
        .threads(4)
        .memoryLimit("2GB")
        .build()
    )

  // MariaDB / MySQL
  val mariaTx =
    Transactor.create(
      MariaConfig
        .builder("localhost", 3306, "mydb", "user", "pass")
        .build()
    )

  // Oracle
  val oracleTx =
    Transactor.create(
      OracleConfig
        .builder("localhost", 1521, "xe", "user", "pass")
        .serviceName("XEPDB1")
        .build()
    )

  // SQL Server — .encrypt(FALSE) for local dev (self-signed cert); use TRUE + trusted cert in prod
  val mssqlTx =
    Transactor.create(
      SqlServerConfig
        .builder("localhost", 1433, "mydb", "user", "pass")
        .encrypt(SqlServerEncrypt.FALSE)
        .build()
    )

  // DB2
  val db2Tx =
    Transactor.create(
      Db2Config
        .builder("localhost", 50000, "mydb", "user", "pass")
        .build()
    )
  // stop
